localstack-core/localstack/services/events/provider.py

```diff
@@ -1364,22 +1366,96 @@ def _process_entry(
                     }
                 )
             )
-            return
+            return {"EventId": event_formatted["id"]}
+
         self._proxy_capture_input_event(event_formatted)
-        if configured_rules := list(event_bus.rules.values()):
-            for rule in configured_rules:
-                self._process_rules(
-                    rule, region, account_id, event_formatted, processed_entries, failed_entry_count
-                )
-        else:
+
+        has_processing_error = self._process_rules_for_event(
+            event_bus.rules.values(), region, account_id, event_formatted, failed_entry_count
+        )
+
+        if has_processing_error:
+            failed_entry_count["count"] += 1
+            return {
+                "ErrorCode": "InternalException",
+                "ErrorMessage": "Error occurred while processing one or more targets",
+            }
+
+        return {"EventId": event_formatted["id"]}
+
+    def _process_rules_for_event(
```

Comment:
FYI we also need to be thoughtful with renaming methods since we patch specific methods in EventStudio and have to always adapt the changes there too (lets minimize renaming)

Reply:
This is a new internal method. used within `_process_entry`. Could this be a problem?